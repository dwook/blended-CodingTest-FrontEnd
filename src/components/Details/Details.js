import React, { Component } from 'react';
import { BASE_URL_POSTSER } from '../../constants';
import './Details.scss';

export default class Details extends Component {
  componentDidMount() {
    this.props.onDetailsLoad(this.props.match.params.id);
  }
  render() {
    const {
      poster_path,
      title,
      overview,
      genres,
      release_date,
      vote_average
    } = this.props.details;
    console.log(this.props.details);
    return (
      Object.keys(this.props.details).length !== 0 && (
        <div className="details">
          <div className="thumbnail">
            <img src={BASE_URL_POSTSER + poster_path} alt={title} />
          </div>
          <div className="info">
            <div className="title">{title}</div>
            <div className="genres">
              {genres.map(item => (
                <span key={item.id}>{item.name}</span>
              ))}
            </div>
            <div className="release_date">{release_date}</div>
            <div className="ratings">
              <span>Ratings</span>
              {vote_average}
            </div>
            <div className="overview">{overview}</div>
          </div>
        </div>
      )
    );
  }
}
