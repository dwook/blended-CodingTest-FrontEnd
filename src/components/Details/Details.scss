.details {
  width: 70%;
  display: flex;
  margin: 0 auto;
  flex-direction: row;
  .thumbnail {
    width: 40%;
    .default {
      display: flex;
      justify-content: center;
      align-items: center;
      width: 80%;
      margin-bottom: 20px;
      min-height: 450px;
      background: rgba(255, 255, 255, 0.2);
      color: #212529;
    }
    img {
      width: 80%;
      margin-bottom: 20px;
    }
  }
  .title {
    width: 90%;
    font-size: 30px;
    margin-bottom: 20px;
    border: none;
    color: #fff;
  }
  .genres {
    width: 90%;
    display: flex;
    flex-wrap: wrap;
    margin: 30px 0 20px;
    border: none;
    span {
      padding: 10px 20px;
      border-radius: 20px;
      margin: 10px 10px 0 0;
      background: #dee2e6;
      color: #212529;
      font-size: 15px;
    }
  }
  .info {
    width: 50%;
    color: #fff;
    .row {
      padding: 15px 0;
      border-bottom: 1px solid rgba(255, 255, 255, 0.5);
    }

    .ratings {
      span {
        margin-right: 20px;
      }
    }
    .overview {
      .title {
        font-size: 20px;
        font-weight: bold;
        margin-bottom: 20px;
      }
    }
    .actors {
      .title {
        font-size: 20px;
        font-weight: bold;
        margin-bottom: 20px;
      }
      .list-people {
        display: flex;
        flex-wrap: wrap;
      }
      .person {
        width: 33%;
        text-align: center;
        .thumbnail {
          width: 100px;
          height: 100px;
          display: flex;
          align-items: center;
          justify-content: center;
          padding: 0;
          margin: 0 auto;
          border-radius: 50%;
          overflow: hidden;
          background: rgba(255, 255, 255, 0.2);
          color: #212529;
          img {
            width: 100%;
          }
        }
        .name {
          margin: 10px 0 25px;
        }
      }
    }
  }
}
