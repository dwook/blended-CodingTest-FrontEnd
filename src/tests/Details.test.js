import React from 'react';
import Details from '../components/Details/Details';
import { shallow } from 'enzyme';

describe('<Details />', () => {
  let wrapper;
  const details = {};
  const actors = [];
  const props = {
    match: {
      params: {
        id: ''
      }
    },
    details: details,
    actors: actors,
    onDetailsLoad: jest.fn()
  };

  beforeEach(() => {
    wrapper = shallow(<Details {...props} />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render details when having no details data', function() {
    expect(wrapper.find('.details').exists()).toEqual(false);
  });

  it('should not render actors list when having no actors', function() {
    expect(wrapper.find('.actors').exists()).toEqual(false);
  });
});
