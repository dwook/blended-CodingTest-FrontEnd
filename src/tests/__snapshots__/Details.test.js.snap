// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> matches snapshot 1`] = `""`;
